src/shared_modules/rsync/CMakeLists.txt

```diff
@@ -6,6 +6,8 @@ if(NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE Release)
 endif()
 
+get_filename_component(SRC_FOLDER ${CMAKE_SOURCE_DIR}/../../ ABSOLUTE)
+
```

Comment:
Please check whether the lib is being compiled for windows and add the target properties statically link libgcc and libstdc++ and also to SUFFIX .dll to the generated lib.